### PR TITLE
[syslog] Add remote syslog configuration (cherry-pick to 202305)

### PR DESCRIPTION
**Pull request by fastiuk**
cherry-pick: #14513
depends: https://github.com/sonic-net/sonic-utilities/pull/2939

* Add an ability to configure remote syslog servers
* Add an initial configuration for remote syslog
* Extend YANG module and add unit tests

#### Why I did it
Adding the following functionality to rsyslog feature:

* Configure remote syslog servers: protocol, filter, severity level
* Update global syslog configuration: severity level, message format

#### How I did it
added parameters to syslog server and global configuration.

#### How to verify it
create syslog server using CLI/adding to Redis-DB
verify server is added to file /etc/rsyslog.conf and server is functional.

#### Description for the changelog
extend rsyslog capabilities, added server and global configuration parameters.

#### Link to config_db schema for YANG module changes
[sonic-syslog.yang](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-syslog.yang)

#### A picture of a cute animal (it is my cat Finn)
PXL_20230413_140941610 PORTRAIT

### CODE REVIEW

File: src/sonic-utilities
Comment (qiluo-msft) [inline src/sonic-utilities:1]: Most of sonic-utilities commits are not relevant. Could you move submodule in a separate PR?

Comment (fastiuk) [reply src/sonic-utilities:1]: Done

**Review by qiluo-msft — APPROVED** [review]

**Comment by fastiuk** [comment]
The fail happens because https://github.com/sonic-net/sonic-utilities/pull/2939 must be merged first.

**Comment by fastiuk** [comment]
@qiluo-msft all checks are passed. It can be merged now

**Comment by StormLiangMS** [comment]
@fastiuk this is new feature, not for 202305, I will revert. 

+ @qiluo-msft  for vis.

**Comment by StormLiangMS** [comment]
revert PR https://github.com/sonic-net/sonic-buildimage/pull/16179

**Comment by fastiuk** [comment]
> @fastiuk This PR is considered as a new feature. Why cherry-pick to 202305? Should we revert?

@qiluo-msft, @StormLiangMS 
That feature was planned for the 202305 release, which is why it was picked into that branch. I don't think we need to revert it.
